### PR TITLE
Fix to trim underlines when errors span lines.

### PR DESCRIPTION
**Pull request by pq**

### CODE REVIEW

**Comment by pq** [comment]
@stereotype441 : this one's for you!

Results now look more like this:

```
/Users/me/src/git/clones/linter/lib/src/scratch.dart 10:3 [lint] PREFER to use /// for doc comments
  /**
  ^^^

1 file analyzed, 1 issue found.
-------------------------------
slash_for_doc_comments        1
-------------------------------
```

File: lib/src/formatter.dart [hunk 1]
```diff
@@ -69,9 +69,10 @@ class DetailedReporter extends SimpleFormatter {
     var contents = getLineContents(line, error);
     out.writeln(contents);
 
-    var arrows = '^' * error.length;
     var spaces = column - 1;
-    var result = '${" " * spaces}$arrows';
+    var arrows = min(error.length, contents.length - spaces);
```
Comment (stereotype441) [inline lib/src/formatter.dart]: Consider changing to:

var arrows = max(1, min(error.length, contents.length - spaces));

just to make sure at least one arrow is always printed.

**Comment by stereotype441** [comment]
LGTM

File: lib/src/formatter.dart (hunk 1 above)
Comment (pq) [inline lib/src/formatter.dart]: Excellent.  Great idea!

**Pull request merged**